Align disk images on 4k sectors

leaberry opened the pull request:
The upstream openwrt images are not aligned on 4k sector boundaries. This leads to qemu throwing errors about lacking the resize permission because the image is opened with O_DIRECT which expects the image to be aligned and will try to align it automatically if it doesn't. But that automatic alignment requires the RESIZE permission in qemu.

During docker image builds we resize the upstream image by a couple MB and specify 'M' to dd instead of 'MB'. The difference is documented in the dd man page but boils down to the difference of 1000*1000 (MB) versus 1024*1024 (M). By using 'M' we ensure an image aligned on 4096 bytes

Additionally this commit modifies the IMAGE_SIZE_ON_INIT code so it also uses 'M' and generates 4k aligned images as well.

Testing: Built a docker image with my changes and ran it

leaberry commented:
I think I worded this commit poorly. What really is being done is making the total image size divisble by 4096 bytes. We aren't actually changing the alignment of partitions inside the image.

AlbrechtL commented:
Thanks for this PR! Outcome of https://github.com/AlbrechtL/openwrt-docker/issues/33.

leaberry commented:
I tested how little size increase I could easily do. The original unzipped image is
```
-rw-r--r--  1 root root 126123520 Jun 23 23:51 openwrt-24.10.2-x86-64-generic-squashfs-combined-efi.img
```
If the dd command is changed to `seek="121"` the resulting image size is an increase of less than 1MB
```
-rw-r--r--  1 root root 126877696 Aug 21 10:38 openwrt-24.10.2-x86-64-generic-squashfs-combined-efi.img
```

I'm going to close this PR, I want to push other changes into my fork and I didn't create a branch for this change. This project is great, thanks for starting/maintaining it.

FWIW I added passthrough folder support so you can define a folder in docker, and qemu will pass it through as a `virtfs` entry which can be mounted with Plan9 (9p) support. I also added support for `LAN_IF="none"` So you can run a complete pci passthrough setup (that's what I do).
